dGame/dComponents/DestroyableComponent.cpp

```diff
@@ -668,44 +678,35 @@ void DestroyableComponent::Damage(uint32_t damage, const LWOOBJID source, uint32
 	}
 
 	//check if hardcore mode is enabled
-    if (Game::config->GetValue("hardcore_mode") == "1") {
+    if (m_HardcoreMode) {
```

Comment:
Ideally this should all be its own function.  This is making damage do more work than it should